test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestExtension.java

```diff
@@ -1198,13 +1197,9 @@ private File getTestOutputDirectory(Class<?> testClass) {
         if (outputDirectory != null) {
             testOutputDirectory = new File(outputDirectory);
         } else {
-            // org.acme.Foo -> org/acme/Foo.class
-            String testClassResourceName = fromClassNameToResourceName(testClass.getName());
-            // org/acme/Foo.class -> /some/path/to/project/target/test-classes/org/acme/Foo.class
-            String testPath = testClass.getClassLoader().getResource(testClassResourceName).toString();
-            // /some/path/to/project/target/test-classes/org/acme/Foo.class -> /some/path/to/project/target/test-classes
-            String testClassesRootPath = testPath.substring(0, testPath.length() - testClassResourceName.length());
-            testOutputDirectory = new File(URI.create(testClassesRootPath));
+            // Gets URL to the root test directory
+            URL testPath = testClass.getClassLoader().getResource(".");
```

Comment:
This is better, but some class loaders will not support getting a directory as a resource, unfortunately.

Reply:
🥲 Good that you know these details better than I do.

Then back to the string formatting it is. I have updated the code comments!